wtransport/src/config.rs

```diff
@@ -63,31 +68,64 @@ impl ServerConfigBuilder<WantsCertificate> {
         self,
         certificate: Certificate,
     ) -> ServerConfigBuilder<WantsTransportConfigServer> {
-        let tls_config = Self::build_tls_config(certificate);
-        let transport_config = TransportConfig::default();
+        let mut tls_config = TlsServerConfig::builder()
+            .with_safe_defaults()
+            .with_no_client_auth()
+            .with_single_cert(certificate.certificates, certificate.key)
+            .unwrap(); // TODO(bfesta): handle this error
+
+        ServerConfig::tweak_tls_config(&mut tls_config);
+
+        self.with_tls_config(tls_config)
+    }
 
+    /// Sets the TLS certificate resolver the server will use for incoming
+    /// WebTransport connections.
+    pub fn with_tls_config(
+        self,
+        tls_config: TlsServerConfig,
+    ) -> ServerConfigBuilder<WantsTransportConfigServer> {
         ServerConfigBuilder(WantsTransportConfigServer {
             bind_address: self.0.bind_address,
             tls_config,
-            transport_config,
-            migration: true,
         })
     }
 
-    fn build_tls_config(certificate: Certificate) -> TlsServerConfig {
-        let mut tls_config = TlsServerConfig::builder()
-            .with_safe_defaults()
-            .with_no_client_auth()
-            .with_single_cert(certificate.certificates, certificate.key)
-            .unwrap(); // TODO(bfesta): handle this error
+    /// Build the [`ServerConfig`] immediately with the low-level
+    /// [`quinn::ServerConfig`].
+    pub fn build(self, quic_config: quinn::ServerConfig) -> ServerConfig {
+        ServerConfig {
+            bind_address: self.0.bind_address,
+            quic_config,
+        }
+    }
+}
 
-        tls_config.alpn_protocols = [WEBTRANSPORT_ALPN.to_vec()].to_vec();
+impl ServerConfigBuilder<WantsTransportConfigServer> {
+    /// Sets the default transport config.
+    pub fn with_default(self) -> ServerConfigBuilder<ReadyServer> {
+        self.with_transport_config(quinn::TransportConfig::default())
+    }
 
-        tls_config
+    /// Sets the transport config to the provided [`quinn::TransportConfig`].
+    pub fn with_transport_config(
```

Comment:
Cannot we just pass the entire QUIC config? Too many "transport" around here :)

I would imagine something like `with_quic_config` (passing `quinn::ServerConfig`)

Reply:
I added https://github.com/BiagioFesta/wtransport/pull/48/files#diff-f7fa9ee5098ae4b2dd53541d64fec3b37057dff18836d9a387acfbd1ea50c164R96-R100 for that. It pretty much short-circuits the building earlier if given the `quic_config`.
I left the transport builder for completion mostly - some users would definitely like this guided flow.